modules/puppet_proxy/puppet_cache.rb

```diff
@@ -19,13 +19,13 @@ def scan_directory_with_cache directory, environment, scanner
         changed    = false
         manifest   = Dir.glob("#{directory}/*").map do |path|
           puppetmodule = File.basename(path)
-          mtime        = File.mtime(path)
+          mtime        = File.mtime(path).to_i.to_s
```

Comment:
I can't think of a disadvantage to doing so, but storing the integer in JSON would be safe enough, no?